### ISSUE TITLE
spiceinit web=yes error; fix not working for me

### ISSUE THREAD

**Issue by ascbot** [DOI-USGS/ISIS3#2315]
---

Author Name: **Robert Herrick** (Robert Herrick)

Original Assignee: Tyler Wilson

---

Hello,

When I try spiceinit from=EW0244228625G.cub (or any other MESSENGER frame) web=yes I get the following error message:
**ERROR** The server sent an unrecognized response.

Seeing the recent thread on this, when I run the fix listed (Tyler Wilson, issue #4552), I get a different set of errors, as follows:
spiceinit from=EW0244228625G.cub url=https://services.isis.astrogeology.usgs.gov/cgi-bin/spiceinit.cgi web=true port=443
QSslSocket: cannot call unresolved function SSLv23_client_method
QSslSocket: cannot call unresolved function SSL_CTX_new
QSslSocket: cannot call unresolved function SSL_library_init
QSslSocket: cannot call unresolved function ERR_get_error
**ERROR** An error occurred when talking to the server. An unknown network-related error occurred.

Any suggestions would be appreciated.

Thx, Robbie Herrick

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Stuart Sides** (@scsides)
Original Date: 2016-12-22T02:01:52Z

---

Robert,
Please post your operating system version.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Robert Herrick** (Robert Herrick)
Original Date: 2016-12-22T02:07:57Z

---

I have a CENTOS box (release 7-2) and am using the RHEL 6, 64-bit version of the current version of ISIS, if that's what you are asking.  By the way, the command was working fine just a few weeks ago.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Tyler Wilson** (Tyler Wilson)
Original Date: 2016-12-22T18:47:16Z

---

Hi Robert, 

Could you paste the results of this command into a message on this ticket?

ldd $ISISROOT/bin/spiceinit

I would like to see which version of the ssl libraries your system is using.  That is where this error is coming from.  

Spiceinit didn't start acting weird until the DOI implemented https migration (a few weeks ago), and we are trying to fix it.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Robert Herrick** (Robert Herrick)
Original Date: 2016-12-22T19:45:43Z

---

 okay, here you go.  I am flying today so internet connection is sporadic.

linux-vdso.so.1 =>  (0x00007ffda8193000)
        libisis3.4.13.so => /usr/local/Isis/isis/bin/../lib/libisis3.4.13.so (0x00007f6060e7a000)
        libQtXmlPatterns.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtXmlPatterns.so.4 (0x00007f6060803000)
        libQtXml.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtXml.so.4 (0x00007f60605c0000)
        libQtNetwork.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtNetwork.so.4 (0x00007f606027b000)
        libQtSql.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtSql.so.4 (0x00007f606003a000)
        libQtGui.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtGui.so.4 (0x00007f605f348000)
        libQtCore.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtCore.so.4 (0x00007f605ee72000)
        libQtSvg.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtSvg.so.4 (0x00007f605ec1a000)
        libQtTest.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtTest.so.4 (0x00007f605e9f5000)
        libQtOpenGL.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtOpenGL.so.4 (0x00007f605e6f7000)
        libQtWebKit.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtWebKit.so.4 (0x00007f605cc7a000)
        libQtDBus.so.4 => /usr/local/Isis/isis/bin/../3rdParty/lib/libQtDBus.so.4 (0x00007f605c9fd000)
        libqwt.so.6 => /usr/local/Isis/isis/bin/../3rdParty/lib/libqwt.so.6 (0x00007f605c70d000)
        libxerces-c-3.1.so => /usr/local/Isis/isis/bin/../3rdParty/lib/libxerces-c-3.1.so (0x00007f605c165000)
        libgeotiff.so.2 => /usr/local/Isis/isis/bin/../3rdParty/lib/libgeotiff.so.2 (0x00007f605bf32000)
        libtiff.so.5 => /usr/local/Isis/isis/bin/../3rdParty/lib/libtiff.so.5 (0x00007f605bcc2000)
        libdsklib.so => /usr/local/Isis/isis/bin/../3rdParty/lib/libdsklib.so (0x00007f60349da000)
        libcspice.so => /usr/local/Isis/isis/bin/../3rdParty/lib/libcspice.so (0x00007f602f081000)
        libgeos-3.3.2.so => /usr/local/Isis/isis/bin/../3rdParty/lib/libgeos-3.3.2.so (0x00007f602eb5c000)
        libgeos_c.so.1 => /usr/local/Isis/isis/bin/../3rdParty/lib/libgeos_c.so.1 (0x00007f602e91d000)
        libgsl.so.0 => /usr/local/Isis/isis/bin/../3rdParty/lib/libgsl.so.0 (0x0000003b24000000)
        libgslcblas.so.0 => /usr/local/Isis/isis/bin/../3rdParty/lib/libgslcblas.so.0 (0x0000003b23c00000)
        libX11.so.6 => /lib64/libX11.so.6 (0x00007f602e5c1000)
        libprotobuf.so.7 => /usr/local/Isis/isis/bin/../3rdParty/lib/libprotobuf.so.7 (0x00007f602e2df000)
        libkdu_a63R.so => /usr/local/Isis/isis/bin/../3rdParty/lib/libkdu_a63R.so (0x00007f602df43000)
        libcholmod.so => /usr/local/Isis/isis/bin/../3rdParty/lib/libcholmod.so (0x00007f602dc4a000)
        libamd.so => /usr/local/Isis/isis/bin/../3rdParty/lib/libamd.so (0x00007f602da41000)
        libcolamd.so => /usr/local/Isis/isis/bin/../3rdParty/lib/libcolamd.so (0x00007f602d83a000)
        liblapack.so => /usr/local/Isis/isis/bin/../3rdParty/lib/liblapack.so (0x00007f602ce6a000)
        libhdf5.so.10 => /usr/local/Isis/isis/bin/../3rdParty/lib/libhdf5.so.10 (0x00007f602c99a000)
        libhdf5_hl.so.10 => /usr/local/Isis/isis/bin/../3rdParty/lib/libhdf5_hl.so.10 (0x00007f602c77b000)
        libhdf5_cpp.so.10 => /usr/local/Isis/isis/bin/../3rdParty/lib/libhdf5_cpp.so.10 (0x00007f602c509000)
        libhdf5_hl_cpp.so.10 => /usr/local/Isis/isis/bin/../3rdParty/lib/libhdf5_hl_cpp.so.10 (0x00007f602c307000)
        libblas.so.3 => /usr/local/Isis/isis/bin/../3rdParty/lib/libblas.so.3 (0x00007f602c0b0000)
        libgfortran.so.3 => /usr/local/Isis/isis/bin/../3rdParty/lib/libgfortran.so.3 (0x00007f602bdbd000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f602bab5000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f602b7b3000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f602b59c000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f602b380000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f602afbe000)
        libz.so.1 => /lib64/libz.so.1 (0x00007f602ada7000)
        libgthread-2.0.so.0 => /lib64/libgthread-2.0.so.0 (0x00007f602aba5000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f602a99d000)
        libglib-2.0.so.0 => /lib64/libglib-2.0.so.0 (0x00007f602a665000)
        libpng12.so.0 => /lib64/libpng12.so.0 (0x00007f602a43e000)
        libfreetype.so.6 => /lib64/libfreetype.so.6 (0x00007f602a198000)
        libgobject-2.0.so.0 => /lib64/libgobject-2.0.so.0 (0x00007f6029f47000)
        libSM.so.6 => /lib64/libSM.so.6 (0x00007f6029d3f000)
        libICE.so.6 => /lib64/libICE.so.6 (0x00007f6029b23000)
        libXrender.so.1 => /lib64/libXrender.so.1 (0x00007f6029918000)
        libfontconfig.so.1 => /lib64/libfontconfig.so.1 (0x00007f60296dc000)
        libXext.so.6 => /lib64/libXext.so.6 (0x00007f60294ca000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f60292c5000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f6061c18000)
        libGL.so.1 => /lib64/libGL.so.1 (0x00007f602902e000)
        libgio-2.0.so.0 => /lib64/libgio-2.0.so.0 (0x00007f6028cb7000)
        libgstapp-0.10.so.0 => /lib64/libgstapp-0.10.so.0 (0x00007f6028aab000)
        libgstinterfaces-0.10.so.0 => /lib64/libgstinterfaces-0.10.so.0 (0x00007f6028899000)
        libgstpbutils-0.10.so.0 => /lib64/libgstpbutils-0.10.so.0 (0x00007f6028674000)
        libgstvideo-0.10.so.0 => /lib64/libgstvideo-0.10.so.0 (0x00007f6028457000)
        libgstbase-0.10.so.0 => /lib64/libgstbase-0.10.so.0 (0x00007f6028203000)
        libgstreamer-0.10.so.0 => /lib64/libgstreamer-0.10.so.0 (0x00007f6027f19000)
        libgmodule-2.0.so.0 => /lib64/libgmodule-2.0.so.0 (0x00007f6027d15000)
        libxml2.so.2 => /lib64/libxml2.so.2 (0x00007f60279ab000)
        libnsl.so.1 => /lib64/libnsl.so.1 (0x00007f6027791000)
        libcurl.so.4 => /lib64/libcurl.so.4 (0x00007f6027529000)
        libjpeg.so.62 => /lib64/libjpeg.so.62 (0x00007f60272d3000)
        libxcb.so.1 => /lib64/libxcb.so.1 (0x00007f60270b1000)
        libffi.so.6 => /lib64/libffi.so.6 (0x00007f6026ea8000)
        libuuid.so.1 => /lib64/libuuid.so.1 (0x00007f6026ca3000)
        libexpat.so.1 => /lib64/libexpat.so.1 (0x00007f6026a78000)
        libglapi.so.0 => /lib64/libglapi.so.0 (0x00007f602684a000)
        libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f6026625000)
        libXdamage.so.1 => /lib64/libXdamage.so.1 (0x00007f6026421000)
        libXfixes.so.3 => /lib64/libXfixes.so.3 (0x00007f602621b000)
        libX11-xcb.so.1 => /lib64/libX11-xcb.so.1 (0x00007f6026019000)
        libxcb-glx.so.0 => /lib64/libxcb-glx.so.0 (0x00007f6025dfe000)
        libxcb-dri2.so.0 => /lib64/libxcb-dri2.so.0 (0x00007f6025bf9000)
        libxcb-dri3.so.0 => /lib64/libxcb-dri3.so.0 (0x00007f60259f6000)
        libxcb-present.so.0 => /lib64/libxcb-present.so.0 (0x00007f60257f2000)
        libxcb-randr.so.0 => /lib64/libxcb-randr.so.0 (0x00007f60255e4000)
        libxcb-xfixes.so.0 => /lib64/libxcb-xfixes.so.0 (0x00007f60253dc000)
        libxcb-render.so.0 => /lib64/libxcb-render.so.0 (0x00007f60251d1000)
        libxcb-shape.so.0 => /lib64/libxcb-shape.so.0 (0x00007f6024fcd000)
        libxcb-sync.so.1 => /lib64/libxcb-sync.so.1 (0x00007f6024dc6000)
        libxshmfence.so.1 => /lib64/libxshmfence.so.1 (0x00007f6024bc2000)
        libXxf86vm.so.1 => /lib64/libXxf86vm.so.1 (0x00007f60249bc000)
        libdrm.so.2 => /lib64/libdrm.so.2 (0x00007f60247af000)
        libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f6024594000)
        liborc-0.4.so.0 => /lib64/liborc-0.4.so.0 (0x00007f6024312000)
        liblzma.so.5 => /lib64/liblzma.so.5 (0x00007f60240ec000)
        libidn.so.11 => /lib64/libidn.so.11 (0x00007f6023eb9000)
        libss##so.1 => /lib64/libss##so.1 (0x00007f6023c8f000)
        libssl3.so => /lib64/libssl3.so (0x00007f6023a4b000)
        libsmime3.so => /lib64/libsmime3.so (0x00007f6023824000)
        libnss3.so => /lib64/libnss3.so (0x00007f60234fe000)
        libnssutil3.so => /lib64/libnssutil3.so (0x00007f60232d1000)
        libplds4.so => /lib64/libplds4.so (0x00007f60230cd000)
        libplc4.so => /lib64/libplc4.so (0x00007f6022ec8000)
        libnspr4.so => /lib64/libnspr4.so (0x00007f6022c89000)
        libgssapi_krb5.so.2 => /lib64/libgssapi_krb5.so.2 (0x00007f6022a3d000)
        libkrb5.so.3 => /lib64/libkrb5.so.3 (0x00007f6022758000)
        libk5crypto.so.3 => /lib64/libk5crypto.so.3 (0x00007f6022525000)
        libcom_err.so.2 => /lib64/libcom_err.so.2 (0x00007f6022321000)
        liblber-2.4.so.2 => /lib64/liblber-2.4.so.2 (0x00007f6022112000)
        libldap-2.4.so.2 => /lib64/libldap-2.4.so.2 (0x00007f6021ebe000)
        libXau.so.6 => /lib64/libXau.so.6 (0x00007f6021cba000)
        libpcre.so.1 => /lib64/libpcre.so.1 (0x00007f6021a58000)
        libssl.so.10 => /lib64/libssl.so.10 (0x00007f60217eb000)
        libcrypto.so.10 => /lib64/libcrypto.so.10 (0x00007f6021402000)
        libkrb5support.so.0 => /lib64/libkrb5support.so.0 (0x00007f60211f3000)
        libkeyutils.so.1 => /lib64/libkeyutils.so.1 (0x00007f6020fee000)
        libsasl2.so.3 => /lib64/libsasl2.so.3 (0x00007f6020dd1000)
        libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007f6020b99000)
        libfreebl3.so => /lib64/libfreebl3.so (0x00007f6020996000)

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Tyler Wilson** (Tyler Wilson)
Original Date: 2016-12-23T20:04:59Z

---

Hi Robert,

The OpenSSL libraries that are causing this error are in /lib64/libssl.so.10.  We do not currently support
the version of CentOS you are using (and will not until the next official release of ISIS in late January), and I suspect that CentOS is using more recent libraries than what the present release of ISIS was compiled against.

I would suggest not using the spiceinit web service for the time being and manually downloading the appropriate spice kernels.  An alternative route would be to install an older version of the OpenSSL libraries that the current release of ISIS will work with.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Robert Herrick** (Robert Herrick)
Original Date: 2017-01-06T21:02:57Z

---

I really need this working fairly urgently as I am desperately trying to generate some data for an LPSC abstract.  I am having multiple problems with workarounds, so any help you can provide would be nice:

1.  If I try to use the isisInstall script (the java Isis installer), then it tells me that I do not have enough space available, regardless of what directory I provide as the data directory, and will not let me continue.  It seems to do a space check based on the space available in the root directory rather than the directory I specify.

2.  I tried downloading CTX spice kernels using the rsync command, but ended up with only data in the "ck" directory and not in the "fk" directory, so spiceinit is not working for ctx.

3.  How would I install older versions of the OpenSSL libraries?

I have spiceinit working for MESSENGER data but really need it also working for CTX and HRSC data.

Thanks,
Robbie

Tyler Wilson wrote:
> Hi Robert,
> 
> The OpenSSL libraries that are causing this error are in /lib64/libssl.so.10.  We do not currently support
> the version of CentOS you are using (and will not until the next official release of ISIS in late January), and I suspect that CentOS is using more recent libraries than what the present release of ISIS was compiled against.
> 
> I would suggest not using the spiceinit web service for the time being and manually downloading the appropriate spice kernels.  An alternative route would be to install an older version of the OpenSSL libraries that the current release of ISIS will work with.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Stuart Sides** (@scsides)
Original Date: 2017-01-10T16:18:43Z

---

Robert Herrick wrote:
> I really need this working fairly urgently as I am desperately trying to generate some data for an LPSC abstract.  I am having multiple problems with workarounds, so any help you can provide would be nice:
> 
> 1.  If I try to use the isisInstall script (the java Isis installer), then it tells me that I do not have enough space available, regardless of what directory I provide as the data directory, and will not let me continue.  It seems to do a space check based on the space available in the root directory rather than the directory I specify.

We will look into the problem where the installer appears to be checking the wrong directory for available space, but this may take some time. A new issue has been created to track this problem #4583
The Java installer does not work for Centos 7.x. We suggest you use the rsync commands.
 
> 
> 2.  I tried downloading CTX spice kernels using the rsync command, but ended up with only data in the "ck" directory and not in the "fk" directory, so spiceinit is not working for ctx.

Did the rsync command finish without any errors?
The MRO mission data area is 134GB. I just checked the rsync servers and both have data in the ck, spk, ik, iak, fk, and sclk directories. 

> 
> 3.  How would I install older versions of the OpenSSL libraries?

If you know how to administer your system, you can look for the older SSL library in the package manager, or get a copy from an older version of CentOS. If the older SSL library is install incorrectly it has to potential to break any applications that are looking for the newer version, so I don't suggest going that route unless you are comfortable with that level of detailed administration. 

> 
> I have spiceinit working for MESSENGER data but really need it also working for CTX and HRSC data.

For you information the MEX data area is 15GB.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Tyler Wilson** (Tyler Wilson)
Original Date: 2017-02-14T19:17:43Z

---

The web server (isis-spice-d) which runs the spice server is using Scientific Linux 6.1 (Carbon release).
This is a Linux version which is no longer supported.  The version of Isis which is running is indeed the most recent production release of Linux (v3.5.0.7383).  Running an ldd command on spiceinit (or any other Isis application) reveals many missing dependendcies.  The fix to this issue requires updating the OS for isis-spice-d to one which is currently supported.   This should fix the issue.   Passing the spiceserver.cgi script the same parameters that spiceinit passes it (but using curl instead to record the output), gives the following:

` Hostname was NOT found in DNS cache
   Trying 137.227.233.136...
 Connected to services.isis.astrogeology.usgs.gov (137.227.233.136) port 443 (#0)
 Initializing NSS with certpath: sql:/etc/pki/nssdb
   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
 SSL connection using TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
 Server certificate:
       subject: CN=services.isis.astrogeology.usgs.gov,O=U.S. Geological Survey,L=Reston,ST=Virginia,C=US
       start date: Dec 05 00:00:00 2016 GMT
       expire date: Dec 13 12:00:00 2017 GMT
       common name: services.isis.astrogeology.usgs.gov
       issuer: CN=DigiCert SHA2 High Assurance Server CA,OU=www.digicert.com,O=DigiCert Inc,C=US
> POST /cgi-bin/spiceinit.cgi HTTP/1.1
> User-Agent: curl/7.37.0
> Host: services.isis.astrogeology.usgs.gov
> Accept: */*
> Content-Length: 59849
> Content-Type: application/x-www-form-urlencoded
> Expect: 100-continue
> 
< HTTP/1.1 100 Continue
< HTTP/1.1 200 OK
< Date: Tue, 14 Feb 2017 19:08:10 GMT
< Content-Length: 0
< Content-Type: text/html; charset=UTF-8
< Strict-Transport-Security:  max-age=31536000
< `

Nothing is returned in the result, but the service is running and there are no problems with connecting to port 443 (as indeed a port scan reveals):

`nmap -v services.isis.astrogeology.usgs.gov

Starting Nmap 6.47 ( http://nmap.org ) at 2017-02-14 12:13 MST
Initiating Ping Scan at 12:13
Scanning services.isis.astrogeology.usgs.gov (137.227.233.136) [2 ports]
Completed Ping Scan at 12:13, 0.02s elapsed (1 total hosts)
Initiating Parallel DNS resolution of 1 host. at 12:13
Completed Parallel DNS resolution of 1 host. at 12:13, 0.13s elapsed
Initiating Connect Scan at 12:13
Scanning services.isis.astrogeology.usgs.gov (137.227.233.136) [1000 ports]
Discovered open port 443/tcp on 137.227.233.136
Discovered open port 80/tcp on 137.227.233.136
Completed Connect Scan at 12:13, 3.96s elapsed (1000 total ports)
Nmap scan report for services.isis.astrogeology.usgs.gov (137.227.233.136)
Host is up (0.026s latency).
Not shown: 998 filtered ports
PORT    STATE SERVICE
80/tcp  open  http
443/tcp open  https

Read data files from: /usr/bin/../share/nmap
Nmap done: 1 IP address (1 host up) scanned in 4.40 seconds`

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Tyler Wilson** (Tyler Wilson)
Original Date: 2017-02-17T20:21:54Z

---

Spiceinit appears to be working now (from inside the USGS network).  I am going to test it tonight from home.  The fix was updating the OS on the web server to CentOS Linux Release 7.3.1611. This necessitated updating the httpd.conf file Apache calls. The new config file needs to be added to the assets directory for spiceserver in the ISIS source tree.   Some third party packages on the server which are graphical in nature needed to be installed as well because Qt libraries depended on them.  I will add a README file in the spiceserver assets with my notes about this problem and how to resolve it in case it shows up again.   This weekend (02/18/2017-02/20/2017) the spice server web service will be down because IT is moving racks around, but it will hopefully be back online by Tuesday (02/21/2017).
